### PR TITLE
add DELETE method

### PR DESCRIPTION
**Pull request by jonas27**

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+	}
+}
+
+// according to rfc guidlines PUT should create or replace resources
+// https://www.rfc-editor.org/rfc/rfc2616#section-9.6
+func (s *server) handlePut(w http.ResponseWriter, r *http.Request, key string) {
+	_, ok := s.db.get(key)
+	if !ok {
+		w.WriteHeader(http.StatusCreated)
```
Comment (avorima) [inline server/server.go]: Only the first WriteHeader call will be used to set the status code. This endpoint will return 201/200 even if an error occured

File: server/main.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+			&db,
+		},
+		server: &http.Server{
+			Addr:              *port,
+			ReadHeaderTimeout: 3 * time.Second,
+		},
+	}
+
+	log.Printf("Server running on port %s\n", *port)
+	return s.server.ListenAndServe()
```
Comment (avorima) [inline server/main.go:40]: The server needs a handler

**Review by avorima — CHANGES_REQUESTED** [review]
Nice to see alternative test runners and libraries. We mainly use [testify](https://github.com/stretchr/testify) and `make`

**Comment by jonas27** [comment]
> Nice to see alternative test runners and libraries. We mainly use [testify](https://github.com/stretchr/testify) and `make`

Haha, its basesically testify in mini, also developed by mat ryer

File: server/main.go (hunk 2 above)
Comment (jonas27) [reply server/main.go:40]: @avorima fixed in #6

File: server/server.go (hunk 1 above)
Comment (jonas27) [reply server/server.go]: @avorima fixed in #15

**Pull request closed without merging**